AlphaWallet/Tokens/Types/TokenObject.swift

```diff
@@ -47,7 +47,10 @@ class TokenObject: Object {
     }
 
     var address: Address {
-        return Address(string: contract)!
+        if let ethAddress = Address(string: contract) {
+            return ethAddress
+        }
+        return Address(uncheckedAgainstNullAddress: Constants.nullAddress)!
```

Comment:
Since the purpose of the PR is to make it safer by not allowing transactions with blacklisted addresses, returning `0x0` on an invalid address can break this promise. In addition, the original version `Address(string: contract)!` before this PR would crash with an invalid address whereas with this PR, it'll be taken as `0x0`.

How about if we do this:

1. Use `Address(uncheckedAgainstNullAddress: contract)!` or keep it as `Address(string: contract)!` here.
2. Have a PR in the future to make as much of the addresses that is now represented as `String`s and turn them into `Address`s so we can pass them around safely (knowing they are at least not blacklisted) and not need to check for nils or force unwrap (similar to what is happening here) after their initial creation? Might even be a good idea to have 2 distinct "address" classes for contract and wallets.. It's Swift after all, we can take advantage of the strong typing :)

What do you think?